### PR TITLE
Basic generation of transfers

### PR DESCRIPTION
**Pull request by datanel**

### CODE REVIEW

**Comment by prhod** [comment]
I tried to run it against Marseilles RTM (not in release mode), but I ended to kill it after 30 min. Is there room for performance improvements ?

**Comment by datanel** [comment]
@prhod  rebased

**Comment by TeXitoi** [comment]
@prhod use release mode, there is in general a 100x speedup.

File: src/transfers.rs [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+                        from_stop_id: sp1.id.clone(),
+                        to_stop_id: sp2.id.clone(),
+                        min_transfer_time: Some(transfer_time),
+                        real_min_transfer_time: Some(transfer_time),
+                        equipment_id: None,
+                    });
+                }
+            }
+        }
+    }
```
Comment (TeXitoi) [inline src/transfers.rs]: Minimizing distance computations.

```rust
let transfers_set = make_transfers_set(&transfers, &stop_points);
let sq_max_distance = max_distance * max_distance;
for (idx1, sp1) in &stop_points {
    let approx = sp1.coord.approx();
    for (_, sp2) in stop_points.values().filter(|&(idx2, _)| !transfers_set.contains((idx1, idx2))) {
        let sq_distance = approx.sq_distance_to(&sp2.coord);
        if sq_distance > sq_max_distance {
            continue;
        }
        let transfer_time = (sq_distance.sqrt() / walking_speed) as u32;
        transfers.push(Transfer {
            from_stop_id: sp1.id.clone(),
            to_stop_id: sp2.id.clone(),
            min_transfer_time: Some(transfer_time),
            real_min_transfer_time: Some(transfer_time),
            equipment_id: None,
        });
    }
}
```

File: src/transfers.rs [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        //we keep the existing transfers
+        assert_eq!(transfers[0].min_transfer_time, Some(50));
+        assert_eq!(transfers[0].real_min_transfer_time, Some(60));
+        assert_eq!(transfers[1].min_transfer_time, Some(200));
+        assert_eq!(transfers[1].real_min_transfer_time, Some(210));
+
+        // transfers sp_2 -> sp_3, sp_3 -> sp_2, sp_3 -> sp_1 are not added,
+        // because distances between them are > 100m
+        // sp_1 -> sp_3 is kept because it is an existing transfer.
+        assert_eq!(6, transfers.len());
```
Comment (TeXitoi) [inline src/transfers.rs]: You should also check the durations of the generated transfers.

File: src/transfers.rs [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        for sp2 in stop_points.values() {
+            if !transfer_exists(&transfers, &sp1, &sp2) {
+                let distance = sp1.coord.distance_to(&sp2.coord);
+                let transfer_time = (distance / walking_speed) as u32;
+                if distance <= max_distance {
+                    transfers.push(Transfer {
+                        from_stop_id: sp1.id.clone(),
+                        to_stop_id: sp2.id.clone(),
+                        min_transfer_time: Some(transfer_time),
+                        real_min_transfer_time: Some(transfer_time),
```
Comment (TeXitoi) [inline src/transfers.rs]: Don't we want different values here?

File: src/transfers.rs (hunk 1 above)
Comment (TeXitoi) [reply src/transfers.rs]: maybe just filtering by distance first before filtering with existing transfers would be enough.

**Pull request merged**